File: No.1_payBackForLoan.m
function [ amount ] = payBackForLoan( P, i, n )
% Calculate the amount of repay for loan
% input:
%   P: The amount of money you loan
%   i: Interest rate of loan
%   n: Years you have to pay back
% output:
%   amount: the amount of money you pay each year

amount = zeros(1,n);
for j = 1:n
    amount(j) = (P*i*(1+i).^j)/((1+i).^j-1);
end
end

File: No.1_payBackForLoan.py
import numpy as np

def payBackForLoan(P, i, n):
    amount = np.zeros(n)
    for j in range(0,n):
        amount[j] = P*i*np.power(1+i,j+1)/(np.power(1+i,j+1)-1)
    return amount
